File: Intervals/35insertIntervals/insertInterval.java

public class insertInterval {
    public static int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> result = new ArrayList<>();
        for (int[] interval : intervals) {
            if (newInterval == null || interval[1] < newInterval[0]) {
                result.add(interval);
            } else if (interval[0] > newInterval[1]) {
                result.add(newInterval);
                result.add(interval);
                newInterval = null;
            } else {
                newInterval[0] = interval[0] < newInterval[0] ? interval[0] : newInterval[0];
                newInterval[1] = interval[1] > newInterval[1] ? interval[1] : newInterval[1];
            }
        }
        if (newInterval != null) {
            result.add(newInterval);
        }
        return result.toArray(new int[result.size()][]);
    }

    public static void main(String[] args) {
        int[][] interval = { { 1, 3 }, { 6, 9 } };
        int[][] result = insert(interval, new int[] { 2, 5 });
        for (int i = 0; i < result.length; i++) {
            System.out.print(Arrays.toString(result[i]));
        }
    }
}

File: Intervals/35insertIntervals/insertInterval.py
class Solution:
    def insertInterval(self, interval: list[list[int]], newInterval: list[int]) -> list[list[int]]:
        result = []
        for i in range(len(interval)):
            if newInterval[1] < interval[i][0]:
                result.append(newInterval)
                return result + interval[i:]
            elif newInterval[0] > interval[i][1]:
                result.append(interval[i])
            else:
                newInterval[0] = min(newInterval[0], interval[i][0])
                newInterval[1] = max(newInterval[1], interval[i][1])
        result.append(newInterval)
        return result
print(Solution().insertInterval([[1,3], [6,9]], [2,5]))